src/Calculator.php

```diff
@@ -8,44 +8,56 @@
 class Calculator
 {
     private ComposerFile $composer;
-    private PackagistAPI $packagist;
+    private PackageAPI $packageAPI;
 
-    public function __construct(ComposerFile $composer, PackagistAPI $packagist)
+    public function __construct(ComposerFile $composer, PackageAPI $packageAPI)
     {
         $this->composer = $composer;
-        $this->packagist = $packagist;
+        $this->packageAPI = $packageAPI;
     }
 
     /**
      * @param string $directory
      * @return Dependency[]
      */
-    public function getDependencyInfo(string $directory): array
-    {
-        $dependencies = $this->composer->getDependencies($directory);
+	public function getDependencyInfo(string $directory): array
+	{
+		$dependencies = $this->composer->getDependencies($directory);
+		$repositories = array_map(
+			fn($repositoryUrl) => $this->packageAPI->getRepositoryInfo($repositoryUrl),
+			$this->composer->getRepositoriesUrl($directory) ?: ['https://repo.packagist.org']
+		);
 
-        foreach ($dependencies as $dependency) {
-            $package_info = $this->packagist->getPackageInfo($dependency->name);
-            if (empty($package_info)) {
-                continue;
-            }
+		foreach ($dependencies as $dependency) {
+			foreach ($repositories as $repository) {
+				if (!$repository->hasPackage($dependency->name)) {
+					continue;
+				}
+				$package_info = $this->packageAPI->getPackageInfo($dependency->name, $repository->getPackageUrl($dependency->name));
+				if (!empty($package_info)) {
+					break;
+				}
+			}
```

Comment:
Could we pull this out into its own `private` method that returns `$package_info` to reduce nesting?